source/language/pulses.rst

```diff
@@ -27,8 +27,11 @@ instruction sequence on *physical* qubits, e.g.
 
    defcal rz(angle[20] theta) $0 { ... }
    defcal measure $0 -> bit { ... }
+   defcal delay[time] $0 { ... }
    defcal measure_iq q -> complex[float[32]] { ... }
 
+defcals may be defined for the qubit ``measure``, ``reset``, ``delay`` operators as
+well as for arbitrarily named gates.
```

Comment:
I think we'll need to expand this to explain much more about the new `[duration]` syntax, for example where it's valid (only `delay`, or any identifier? - without speaking for everyone, my impression is that there's only support in the TSC for `delay` supporting the time argument in 3.0, nothing else).

There's also a question of what the definition should look like here.  In `defcal`s of parametrised gates, there's two options for a parameter definition - it can either be an `expression` (which must evaluate to a type of `const angle[_]`), or it's a `type identifier` declaration (like `angle[12] theta`).  I think there's a decision that needs to be made here along similar lines:

1. have the only allowed thing be `"[" identifier "]"`, which causes `identifier` to be a parameter that's implicitly of type `duration` within the `defcal` body (subject to however that crosses into the particular pulse grammar)
2. have this be `"[" "duration" identifier "]"`, with the type explicit.
3. the same as point 2, but also allow `"[" expression "]"`, where `expression` must evaluate to a type `const duration`.

Form 1 is what's currently proposed, but may well be at odds with how types are explicit in (almost) every other part of the language, include the other parts of `defcal` statements.  Form 2 is perhaps more consistent with how parameters are usually declared.  Form 3 would perhaps be the most similar to the other parameters in `defcal`s are declared, and would permit things like specifying a `defcal delay[10ms]` that's different at compile time to a more general`defcal delay[duration t]` - not so useful for `delay`, but makes more sense in the context of future extensions of this syntax to gates.

(My immediate preference in OQ 3.0 is for form 2, but it's not my decision - it's for the TSC and for other implementors to weigh in on.)

Reply:
@PhilReinhold can you please update this PR in light of above discussion?

Reply:
Thanks, Phil